pkg/discovery/backends.go

```diff
@@ -39,6 +40,70 @@ func parse(rawurl string) (string, string) {
 	return parts[0], parts[1]
 }
 
+// ParseAdvertise parses the --cluster-advertise daemon config which accepts
+// <ip-address>:<port> or <interface-name>:<port>
+func ParseAdvertise(store, advertise string) (string, error) {
+	var (
+		iface *net.Interface
+		addrs []net.Addr
+		err   error
+	)
+
+	addrPort := strings.Split(advertise, ":")
```

Comment:
@cpuguy83 @calavera fixed it. this also addresses @cpuguy83 another issue on forcing the :port during the configuration instead of failing during discovery.